### PR TITLE
Upgrade to AWS SDK 1.11.749

### LINKED ISSUE #53191: Upgrade repository-s3 plugin to use aws-sdk-java library version 1.11.704 or greater
**Elasticsearch version** (`bin/elasticsearch --version`): 7.6

**Plugins installed**: [`repository-s3`]

**Description of the problem including expected versus actual behavior**:

The aws-sdk-java library (version 1.11.636) used in the s3-repository plugin contains a [bug](https://github.com/aws/aws-sdk-java/issues/2136) that prevents the IAM container role on Amazon EKS deployments from authenticating to s3 buckets. The IAM container role authentication bug is resolved as of 1.11.704.

The request here is to use a version of the aws-sdk-java library that is equal to or greater than 1.11.704 in the `repository-s3` plugin.

### PR DESCRIPTION
**Pull request by original-brownbear**
Upgrading AWS SDK to v1.11.749.
Required building clients inside privileged contexts because some class loading that requires privileges now happens there
and working around a new SDK bug in the S3 client builder.

Closes #53191

### CODE REVIEW

File: plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Service.java [hunk 1]
```diff
@@ -141,7 +141,12 @@ AmazonS3 buildClient(final S3ClientSettings clientSettings) {
         builder.withCredentials(buildCredentials(logger, clientSettings));
         builder.withClientConfiguration(buildConfiguration(clientSettings));
 
-        final String endpoint = Strings.hasLength(clientSettings.endpoint) ? clientSettings.endpoint : Constants.S3_HOSTNAME;
+        String endpoint = Strings.hasLength(clientSettings.endpoint) ? clientSettings.endpoint : Constants.S3_HOSTNAME;
+        if ((endpoint.startsWith("http://") || endpoint.startsWith("https://")) == false) {
+            // Manually add the schema to the endpoint to work around https://github.com/aws/aws-sdk-java/issues/2274
+            // TODO: Remove this once fixed in the AWS SDK
+            endpoint = clientSettings.protocol.toString() + "://" + endpoint;
```
Comment (tlrx) [inline plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Service.java]: :(

**Review by tlrx — APPROVED** [review]
LGTM, thanks Armin

**Comment by original-brownbear** [comment]
Thanks Tanguy!

**Comment by JackCloudman** [comment]
hello, i have same problem with version 7.8

**Pull request merged**